testing/pow_test.py

```diff
@@ -0,0 +1,69 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+assert 2.0 ** -2 == 0.25, "2.0 ** -2"
+assert 2.0 ** -1 == 0.5, "2.0 ** -1"
+assert 2.0 ** 0 == 1, "2.0 ** 0"
+assert 2.0 ** 1 == 2, "2.0 ** 1"
+assert 2.0 ** 2 == 4, "2.0 ** 2"
+
+assert (-2.0) ** -2 == 0.25, "(-2.0) ** -2"
+assert (-2.0) ** -1 == -0.5, "(-2.0) ** -1"
+assert (-2.0) ** 0 == 1, "(-2.0) ** 0"
+assert (-2.0) ** 1 == -2, "(-2.0) ** 1"
+assert (-2.0) ** 2 == 4, "(-2.0) ** 2"
+
+assert 2 ** -2 == 0.25, "2 ** -2"
+assert 2 ** -1 == 0.5, "2 ** -1"
+assert 2 ** 0 == 1, "2 ** 0"
+assert 2 ** 1 == 2, "2 ** 1"
+assert 2 ** 2 == 4, "2 ** 2"
```

Comment:
Good to have some explicit long tests here, e.g. `assert 2L ** 2 == 4, "2L ** 2"`